src/scripts/utils/tokens.ts

```diff
@@ -1,63 +1,94 @@
 import { Contract } from '@ethersproject/contracts'
 import { JsonRpcProvider } from '@ethersproject/providers'
+import { } from '@ethersproject/strings'
 import { ERC20_ABI } from "../../abi/erc20";
 import type { TokenInfo } from '@uniswap/token-lists'
+import { readCsv } from './csv'
 
-
-function handleErrorReturnEmptyString(error: any) {
-  console.error(error)
-  return ''
+interface ReturnCall {
+  success: boolean
+  returnData: string
 }
 
 export interface Erc20Info {
-  symbol: string
-  decimals: number
-  name: string
+  symbol?: string
+  decimals?: number
+  name?: string
+}
+interface Call {
+  target: string
+  callData: string
 }
 
-export async function getTokenInfos(tokenAddresses: string[], _multicall: Contract, provider: JsonRpcProvider): Promise<Map<string, Erc20Info>> {
+
+export async function getTokenInfos(tokenAddresses: string[], multicall: Contract, provider: JsonRpcProvider): Promise<Map<string, Erc20Info>> {
   const tokenInfos = new Map<string, Erc20Info>();
-  let i = 0
-  for (const address of tokenAddresses) {
-    console.log('Get token info for ', address, i++)
-    const erc20Contract = new Contract(address, ERC20_ABI, provider)
-    const symbol = await erc20Contract.symbol().catch(handleErrorReturnEmptyString)
-    const decimals = await erc20Contract.decimals().catch((e: any) => {
-      console.error('Error getting decimals for ', address, e)
-      return 18
+
+  if (tokenAddresses.length === 0) {
+    return tokenInfos
+  }
+
+  const erc20Contract = new Contract(tokenAddresses[0], ERC20_ABI, provider)
```

Comment:
At first, I was confused why the contract is created with the first token address?
But then I got why.
Anyway, we could move the interface to a global const:

```
export const ERC20_INTERFACE = new Contract('', ERC20_ABI).interface
```